### PR TITLE
Feat: user role environment assignment

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of #654 

Added user - environment assignment. (team assignment will follow up).

1. Updated DELETE rest call to allow params (required for the new API calls).
2. Added assignment API calls.
3. Added unit tests for new API calls.
4. Added resource schema.
5. Added acceptance tests to new resource.
6. Added integration tests.
7. Added examples.

### CODE REVIEW

File: examples/resources/env0_user_environment_assignment/resource.tf [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+resource "env0_api_key" "api_key" {
+  name              = "user-api-key-sample"
+  organization_role = "User"
+}
+
+resource "env0_custom_role" "custom_role" {
+  name = "custom-role-sample"
+  permissions = [
+    "VIEW_PROJECT",
```
Comment (yarivg) [inline examples/resources/env0_user_environment_assignment/resource.tf:23]: I would change the given permission to environment permissions to be more real

File: env0/resource_user_environment_assignment.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	return &schema.Resource{
+		CreateContext: resourceUserEnvironmentAssignmentCreate,
+		UpdateContext: resourceUserEnvironmentAssignmentUpdate,
+		ReadContext:   resourceUserEnvironmentAssignmentRead,
+		DeleteContext: resourceUserEnvironmentAssignmentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:        schema.TypeString,
+				Description: `id of the user. Note: can also be an id of a "User" API key`,
```
Comment (yarivg) [inline env0/resource_user_environment_assignment.go:22]: are you sure about the note? haven't tried it with 'user' api-keys

**Review by yarivg — COMMENTED** [review]
Went over most of it and left some comments.. not approving since I don't know go @yaronya 
@TomerHeber

**Review by yarivg — APPROVED** [review]
The new API for assignment of role to a user on environment seems to be written well
👍

File: env0/resource_user_environment_assignment.go (hunk 2 above)
Comment (TomerHeber) [reply env0/resource_user_environment_assignment.go:22]: I'll remove the comment.

**Pull request merged**